### PR TITLE
REF: combine all alignment into _align_method_FRAME

### PR DESCRIPTION
**Pull request by jbrockmendel**
This puts us within striking distance of de-duplicating _flex_comp_method_FRAME and _comp_method_FRAME, with _arith_method_FRAME not far behind

### CODE REVIEW

File: pandas/core/ops/__init__.py [hunk 1]
```diff
@@ -609,8 +609,25 @@ def _combine_series_frame(left, right, func, axis: int):
...
+    Convert rhs to meet lhs dims if input is list, tuple or np.ndarray.
+
+    Parameters
+    ----------
+    left : DataFrame
+    right : Any
+    axis: int, str, or None
+    flex: bool or None, default False
+        Whether this is a flex op, in which case we reindex.
+        None indices not to check for alignment.
```
Comment (WillAyd) [inline pandas/core/ops/__init__.py]:
```suggestion
        None indicates not to check for alignment.
```

I *think* this is a typo

File: pandas/core/ops/__init__.py [hunk 2]
```diff
@@ -609,8 +609,25 @@ def _combine_series_frame(left, right, func, axis: int):
     return left._construct_result(new_data)
 
 
-def _align_method_FRAME(left, right, axis):
-    """ convert rhs to meet lhs dims if input is list, tuple or np.ndarray """
+def _align_method_FRAME(left, right, axis, flex=False, level=None):
```
Comment (WillAyd) [inline pandas/core/ops/__init__.py]:
```suggestion
def _align_method_FRAME(left, right, axis, flex: Optional[bool] = False, level: Optional[Label] = None):
```

I think should add types for the new arguments here

**Review by WillAyd — CHANGES_REQUESTED** [review]
looks good generally minor things

File: pandas/core/ops/__init__.py [hunk 3]
```diff
@@ -609,8 +609,25 @@ def _combine_series_frame(left, right, func, axis: int):
     return left._construct_result(new_data)
 
 
-def _align_method_FRAME(left, right, axis):
-    """ convert rhs to meet lhs dims if input is list, tuple or np.ndarray """
+def _align_method_FRAME(left, right, axis, flex: Optional[bool] = False, level: Optional[Label] = None):
```
Comment (WillAyd) [inline pandas/core/ops/__init__.py]:
```suggestion
def _align_method_FRAME(left, right, axis, flex: Optional[bool] = False, level: Optional[Level] = None):
```

Sorry typo on previous; we actually have Level defined in pandas._typing already (probably need to import)

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**